cms/djangoapps/contentstore/utils.py

```diff
@@ -575,7 +575,7 @@ def get_sibling_urls(subsection):
             # won't display a link to a previous unit.
         else:
             try:
-                prev_section = sections[sections.index(section) - 1]
+                prev_section = sections[sections.index([s for s in sections if s.location == section.location][0]) - 1]
```

Comment:
This is needed because unlike XModule subclasses, XBlocks do not have `__eq__()`.

Reply:
Tiniest of nits: if you're just grabbing the first thing and throwing out the other values, you can also use `next`:

```python
prev_section = sections[sections.index(next(s for s in sections if s.location == section.location)) - 1]
```

Reply:
Will update when doing the rebase.

Reply:
Have updated this. Thanks for the suggestion!